### ISSUE TITLE
Why do you use 'torch.clamp(classification, 1e-4, 1.0 - 1e-4)' in loss function?

### ISSUE THREAD

**Issue by q1nwu** [yhenon/pytorch-retinanet#3]
Hi, yhenon. Thanks for you share your code.
I found this code 'torch.clamp(classification, 1e-4, 1.0 - 1e-4)' in loss function. I don't know why we should clamp the classification results. 
Thanks very much.

**Comment by yhenon** [comment]
Hi,
I'm glad you find this code useful. We clamp the classification results when calculating the loss for numerical reasons.
Inside the loss, we calculate `torch.log(classification)`. If any value inside of `classification` is equal to zero, then we take log(0), which is not valid, and results in `NaN`.
I hope this answers your question.
